src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -92,6 +92,54 @@ internal static class ProducerConsumer<TItem>
             cancellationToken);
     }
 
+    /// <summary>
+    /// Version of RunAsync that will process <paramref name="source"/> in parallel.
+    /// </summary>
+    public static Task RunParallelAsync<TSource, TArgs>(
```

Comment:
>RunParallelAsync [](http://example.com/codeflow?start=23&length=16)

:thumbsup:

Reply:
just a though, could runParallel be an option?

Reply:
well, it's a very different sig.  it needs the stream of data to process.

or did you mean that you could pass in the stream of data, but ask to process it linearly instead, based on the value of 'runParallel'?

Reply:
This question was posed without reading the PR. It probably has no merit, but I'll read through this a bit more in the morning, and then likely edit the comment to make it sound more intelligent. :)

Reply:
sgtm :)